back-end/java/main/java/src/DormRoom/DormBuilding.java

```diff
@@ -0,0 +1,192 @@
+   * associated details (e.g., geographical coordinates, construction year, and laundry ratio).
+   *
+   * @param buildingName the official name of the building (e.g. "Barbour Hall", "Buxton House")
+   * @throws RuntimeException if the building name does not match any known buildings in the predefined mapping
+   */
+  public DormBuilding(String buildingName) {
+    this.buildingName = buildingName;
+
+    switch (buildingName) {
+      case "Barbour Hall" -> {
+        this.peoplePerWasher = 56f;
+        this.year = 1904;
+        this.longitude = -71.398025;
+        this.latitude = 41.823856;
+        this.campusLocation = CampusLocation.EastCampus;
+      }
+      case "Buxton House" -> {
+        this.peoplePerWasher = 0f;
+        this.year = 1951;
+        this.longitude = -71.402334;
+        this.latitude = 41.824522;
+        this.campusLocation = CampusLocation.WristonQuad;
+      }
+      case "Caswell Hall" -> {
+        this.peoplePerWasher = 47f;
+        this.year = 1903;
+        this.longitude = -71.400623;
+        this.latitude = 41.826106;
+        this.campusLocation = CampusLocation.SimmonsQuad;
+      }
+      case "Chapin House" -> {
+        this.peoplePerWasher = 38.33333333f;
+        this.year = 1951;
+        this.longitude = -71.400398;
+        this.latitude = 41.82446;
+        this.campusLocation = CampusLocation.WristonQuad;
+      }
+      case "Diman House" -> {
+        this.peoplePerWasher = 29.5f;
+        this.year = 1951;
+        this.longitude = -71.401556;
+        this.latitude = 41.82439;
+        this.campusLocation = CampusLocation.WristonQuad;
+      }
+      case "Goddard House" -> {
+        this.peoplePerWasher = 39.33333333f;
+        this.year = 1951;
+        this.longitude = -71.401293;
+        this.latitude = 41.824126;
+        this.campusLocation = CampusLocation.WristonQuad;
+      }
+      case "Grad Center A" -> {
+        this.peoplePerWasher = 36.66666667f;
+        this.year = 1968;
+        this.longitude = -71.400953;
+        this.latitude = 41.823065;
+        this.campusLocation = CampusLocation.GradCenter;
+      }
+      case "Grad Center B" -> {
+        this.peoplePerWasher = 38.33333333f;
+        this.year = 1968;
+        this.longitude = -71.401009;
+        this.latitude = 41.823566;
+        this.campusLocation = CampusLocation.GradCenter;
+      }
+      case "Grad Center C" -> {
+        this.peoplePerWasher = 36.66666667f;
+        this.year = 1968;
+        this.longitude = -71.400368;
+        this.latitude = 41.823628;
+        this.campusLocation = CampusLocation.GradCenter;
+      }
+      case "Grad Center D" -> {
+        this.peoplePerWasher = 38f;
+        this.year = 1968;
+        this.longitude = -71.400305;
+        this.latitude = 41.823124;
+        this.campusLocation = CampusLocation.GradCenter;
+      }
+      case null, default -> throw new RuntimeException("Not a valid building name: " + buildingName);
```

Comment:
Should this be case default? Im not familiar with case null, nut it seems like only a null input triggers this case. If an invalid name is given what happens

Reply:
This code signifies null or default resulting in an Exception.